src/lib/tls/tls_server.h

```diff
@@ -20,13 +20,22 @@ namespace TLS {
 
 class Server_Handshake_State;
 
+struct BOTAN_UNSTABLE_API DTLS_Prestate final
+   {
+   public:
```

Comment:
Instead of struct this should be a proper class with constructors [2, one for new prestate and the other for accepted prestate] and an accessor for cookie_valid, plus a `friend` decl for Channel so it can get at the internal values.